app/bundles/LeadBundle/EventListener/DoNotContactSubscriber.php

```diff
@@ -0,0 +1,65 @@
+<?php
+
+declare(strict_types=1);
+
+/*
+ * @copyright   2014 Mautic Contributors. All rights reserved
+ * @author      Mautic
+ *
+ * @link        http://mautic.org
+ *
+ * @license     GNU/GPLv3 http://www.gnu.org/licenses/gpl-3.0.html
+ */
+
+namespace Mautic\LeadBundle\EventListener;
+
+use Mautic\LeadBundle\Event\DoNotContactAddEvent;
+use Mautic\LeadBundle\Event\DoNotContactRemoveEvent;
+use Mautic\LeadBundle\Model\DoNotContact;
+use Symfony\Component\EventDispatcher\EventSubscriberInterface;
+
+class DoNotContactSubscriber implements EventSubscriberInterface
```

Comment:
I can't find a reason why this class should be extendable. Subscribers aren't used as dependencies of other services so there is no need to mock them.
```suggestion
final class DoNotContactSubscriber implements EventSubscriberInterface
```